src/main/java/net/sf/mzmine/modules/peaklistmethods/alignment/adap3/ADAP3AlignerTask.java

```diff
@@ -0,0 +1,419 @@
+import net.sf.mzmine.datamodel.RawDataFile;
+import net.sf.mzmine.datamodel.impl.SimplePeakInformation;
+import net.sf.mzmine.datamodel.impl.SimplePeakList;
+import net.sf.mzmine.datamodel.impl.SimplePeakListRow;
+import net.sf.mzmine.modules.peaklistmethods.qualityparameters.QualityParameters;
+import net.sf.mzmine.parameters.ParameterSet;
+import net.sf.mzmine.taskcontrol.AbstractTask;
+import net.sf.mzmine.taskcontrol.TaskStatus;
+
+/**
+ * @author aleksandrsmirnov
+ */
+public class ADAP3AlignerTask extends AbstractTask {
+
+    private static final Logger LOG = Logger.getLogger(
+            ADAP3AlignerTask.class.getName());
+
+    private final MZmineProject project;
+    private final ParameterSet parameters;
+
+    private final PeakList[] peakLists;
+
+    private final String peakListName;
+
+    private final Project alignment;
+
+    public ADAP3AlignerTask(MZmineProject project, ParameterSet parameters) {
+
+        this.project = project;
+        this.parameters = parameters;
+
+        this.peakLists = parameters.getParameter(
+                ADAP3AlignerParameters.PEAK_LISTS)
+                .getValue().getMatchingPeakLists();
+
+        this.peakListName = parameters.getParameter(
+                ADAP3AlignerParameters.NEW_PEAK_LIST_NAME).getValue();
+
+        this.alignment = new Project();
+    }
+
+    @Override
+    public String getTaskDescription() {
+        return "ADAP Aligner, " + peakListName + " (" + peakLists.length
+                + " peak lists)";
+    }
+
+    @Override
+    public double getFinishedPercentage() {
+        return alignment.getProcessedPercent();
+    }
+
+    @Override
+    public void run() {
+
+        if (!isCanceled()) {
+            String errorMsg = null;
+
+            setStatus(TaskStatus.PROCESSING);
+            LOG.info("Started ADAP Peak Alignment");
+
+            try {
+                PeakList peakList = alignPeaks(peakLists);
+
+                if (!isCanceled()) {
+                    project.addPeakList(peakList);
+
+                    QualityParameters.calculateQualityParameters(peakList);
+
+                    setStatus(TaskStatus.FINISHED);
+                    LOG.info("Finished ADAP Peak Alignment");
+                }
+            } catch (IllegalArgumentException e) {
+                errorMsg = "Incorrect Peak Lists:\n" + e.getMessage();
+            } catch (Exception e) {
+                errorMsg = "'Unknown error' during alignment. \n"
+                        + e.getMessage();
+            } catch (Throwable t) {
+                setStatus(TaskStatus.ERROR);
+                setErrorMessage(t.getMessage());
+                LOG.log(Level.SEVERE, "ADAP Alignment error", t);
+            }
+
+            // Report error
+            if (errorMsg != null) {
+                setErrorMessage(errorMsg);
+                setStatus(TaskStatus.ERROR);
+            }
+        }
+    }
+
+    private PeakList alignPeaks(PeakList[] peakLists) {
```

Comment:
the parameter peakLists is redundant, because you already have a private field peakLists